Find a functionally equivalent code:android.util.SparseIntArray.put:COMMENT
Method Modifier: public      
Comment:/**
 * Adds a mapping from the specified key to the specified value,
 * replacing the previous mapping from the specified key if there
 * was one.
 */

@@ -4,26 +4,11 @@
         mValues[i] = value;
     } else {
         i = ~i;
-        if (mSize >= mKeys.length) {
-            int n = ArrayUtils.idealIntArraySize(mSize + 1);
-            int[] nkeys = new int[n];
-            int[] nvalues = new int[n];
-            // Log.e("SparseIntArray", "grow " + mKeys.length + " to " + n);
-            System.arraycopy(mKeys, 0, nkeys, 0, mKeys.length);
-            System.arraycopy(mValues, 0, nvalues, 0, mValues.length);
-            mKeys = nkeys;
-            mValues = nvalues;
-        }
-        if (mSize - i != 0) {
-            // Log.e("SparseIntArray", "move " + (mSize - i));
-            System.arraycopy(mKeys, i, mKeys, i + 1, mSize - i);
-            System.arraycopy(mValues, i, mValues, i + 1, mSize - i);
-        }
-        mKeys[i] = key;
-        mValues[i] = value;
+        mKeys = GrowingArrayUtils.insert(mKeys, mSize, i, key);
+        mValues = GrowingArrayUtils.insert(mValues, mSize, i, value);
         mSize++;
     }
 }

